### DISCUSSION TITLE
Cool idea, mind guiding me on how to rewrite this in Go? What would I need in order to build some of the existing `spkg`s?

### DISCUSSION THREAD

**Discussion by xplshn** [smileaf419/spkg#35] [Ideas]
Hi, awesome project. Would you be interested in guiding me on how to use the SPKG repo? Also, what about writting a spec and using a format like YAML, something that is both human readable and also parsable... It'd be neat. I currently maintain https://github.com/xplshn/dbin, but I'm looking to start a new project, since I need a system-wide package manager for my DIY Linux distro capable of managing files under `/etc`, installing & uninstalling kernels, etc.

Would you be interested in this?

You can join me here, on this group, more specifically, in the `dbin` room in the chat :)

https://t.me/official_loonix/3966

**Comment by smileaf419** [comment]
Never used YAML.
build files are written in simple bash as the script itself just sources them within a subshell.
I have plans of allowing custom repositories but currently this isn't implemented.

**Reply by xplshn** [reply]
I was thinking something more maintainable long-term, and with the option to build/distribute/install/remove binary packages too

**Reply by smileaf419** [reply]
There is an option for installing an Archive (using -A or --installArchive), atm however it doesn't have the ability to fetch the archive file as I don't have a server to host the files, this sorta got put on the back burner.
As for long-term maintainable, in all honesty this is basically a custom rewrite of Gentoo's Portage which is over 20yrs old. So its a very maintainable concept.

**Reply by xplshn** [reply]
I don't think a server is needed. You can take a look at my package manager, self-contained executable format, and repo & store for it, I don't have a server either, but you can still do a lot of this stuff without a server.

Also, if it wasn't clear, I want to make a package manager, inspired in your works, however, since its a lot of shell, and no parsable format is used, it could prove difficult...

I'd like to make a runner that builds these recipes as packages (using Go)
After that, I'd like to make a package manager like `dbin` but to manage these packages

https://github.com/xplshn/AppBundleHUB - Repo & Store
https://github.com/xplshn/dbin - package manager with 3093 (portable) packages in its repos
https://github.com/xplshn/pelf - portable executable format that is self-contained

**Reply by smileaf419** [reply]
The build files themselves contain all the information you'll need broken up into steps. Which should be plenty readable.
one goal of mine was to keep these steps as "raw" as I could as one thing I don't like about gentoo's ebuilds are they use so many special commands its difficult to know whats actually going on. making writing them that much more difficult without deep diving into their code to know what I wanted.
If you need to know what a special command is doing reference pkg-utils and builddefaults

**Comment by smileaf419** [comment]
If you're looking to write build files take a look at the wiki, it should be mostly up to date.
https://github.com/smileaf419/spkg/wiki/Build-Files
if your looking to make your own repository, the overlay I had mentioned previously isn't in, but if you wanted to edit and keep a separate one of your own, the only reference to my repository is within the bootstrap file.
I don't use any social media really apart from discord, not figured out how to do a perm link yet, but this one will last for 7 days. https://discord.gg/m3KR6wYk

**Reply by xplshn** [reply]
oh, okay. I don't know how to use Discord, and I don't have a phone with which to open an account (asks for SMS verification), so, too bad then.
